spec/custom-gutter-component-spec.coffee

```diff
@@ -0,0 +1,132 @@
+CustomGutterComponent = require '../src/custom-gutter-component'
+Gutter = require '../src/gutter'
+
+describe "CustomGutterComponent", ->
+  [customGutterComponent, gutter] = []
+
+  beforeEach ->
+    mockGutterContainer = {}
+    gutter = new Gutter(mockGutterContainer, {name: 'test-gutter'})
+    customGutterComponent = new CustomGutterComponent({gutter})
+
+  it "creates a gutter DOM node with only an empty 'custom-decorations' child node when it is initialized", ->
+    expect(customGutterComponent.getDomNode().classList.contains('gutter')).toBe true
+    expect(customGutterComponent.getDomNode().getAttribute('gutter-name')).toBe 'test-gutter'
+    expect(customGutterComponent.getDomNode().children.length).toBe 1
+    decorationsWrapperNode = customGutterComponent.getDomNode().children.item(0)
+    expect(decorationsWrapperNode.classList.contains('custom-decorations')).toBe true
+
+  it "makes its view accessible from the view registry", ->
+    expect(customGutterComponent.getDomNode()).toBe atom.views.getView(gutter)
+
+  it "hides its DOM node when ::hideNode is called, and shows its DOM node when ::showNode is called", ->
+    customGutterComponent.hideNode()
+    expect(customGutterComponent.getDomNode().style.display).toBe 'none'
+    customGutterComponent.showNode()
+    expect(customGutterComponent.getDomNode().style.display).toBe ''
+
+  describe "::updateSync", ->
+    decorationItem1 = document.createElement('div')
+
+    buildTestState = (customDecorations) ->
```

Comment:
This is pretty cool, but it seems like you could integration test this against the real presenter and have both better coverage and a simpler test. Thoughts?